src/client.rs

```diff
@@ -347,8 +377,11 @@ impl<T: Read + Write> Client<T> {
         username: U,
         password: P,
     ) -> ::std::result::Result<Session<T>, (Error, Client<T>)> {
-        let u = ok_or_unauth_client_err!(validate_str(username.as_ref()), self);
-        let p = ok_or_unauth_client_err!(validate_str(password.as_ref()), self);
+        let synopsis = "LOGIN username password";
```

Comment:
It's really weird to me that the synopsis always includes the argument list, and then the argument is repeated again. I would prefer to have synopsis just be "command", and then emit:

> Invalid character {:?} in argument 'username' of command 'LOGIN'

As long as the parameter name matches the name of an argument to a function in `imap`, I think users will be able to backtrack where the error occurred without us making the errors needlessly verbose.

Reply:
I see your point. I have removed the arguments from the "synopsis".